### PR TITLE
MainWindow.cpp: Improved code for flarm radar positions

### PR DESCRIPTION
**Pull request by Joerg-CH**
Your PR should:
  * all description should be in the commit messages (no text in the pr)
  * your pr should be rebased on the current HEAD
  * one commit per atomic feature (every commit should build)
  * no fixup commits
  * pass all the compile and CI targets

For coding, style guide, architecture information please see our development guide:
https://xcsoar.readthedocs.io/en/latest/index.html

### CODE REVIEW

**Comment by Joerg-CH** [comment]
This will solve issue  #1575

File: src/MainWindow.cpp [hunk 1]
```diff
@@ -430,38 +430,42 @@ MainWindow::ReinitialiseLayout_flarm(PixelRect rc,
...
   case TrafficSettings::GaugeLocation::TOP_RIGHT_AVOID_IB:
     rc.top = GetMainRect().top;
+	rc.right = GetMainRect().right;
     rc.left = rc.right - width;
     rc.bottom = rc.top + height;
     break;
 
   case TrafficSettings::GaugeLocation::BOTTOM_LEFT_AVOID_IB:
     rc.bottom = GetMainRect().bottom;
+	rc.left = GetMainRect().left;
```
Comment (lordfolken) [inline src/MainWindow.cpp:447]: Please remove indent.

File: src/MainWindow.cpp [hunk 2]
```diff
@@ -430,38 +430,42 @@ MainWindow::ReinitialiseLayout_flarm(PixelRect rc,
...
   case TrafficSettings::GaugeLocation::TOP_LEFT_AVOID_IB:
     rc.top = GetMainRect().top;
+	rc.left = GetMainRect().left;
     rc.right = rc.left + width;
     rc.bottom = rc.top + height;
     break;
 
   case TrafficSettings::GaugeLocation::TOP_RIGHT_AVOID_IB:
     rc.top = GetMainRect().top;
+	rc.right = GetMainRect().right;
```
Comment (lordfolken) [inline src/MainWindow.cpp:440]: Please remove indent.

File: src/MainWindow.cpp [hunk 3]
```diff
@@ -430,38 +430,42 @@ MainWindow::ReinitialiseLayout_flarm(PixelRect rc,
 
   case TrafficSettings::GaugeLocation::TOP_LEFT_AVOID_IB:
     rc.top = GetMainRect().top;
+	rc.left = GetMainRect().left;
```
Comment (lordfolken) [inline src/MainWindow.cpp:433]: Please remove indent.

File: src/MainWindow.cpp [hunk 4]
```diff
@@ -430,38 +430,42 @@ MainWindow::ReinitialiseLayout_flarm(PixelRect rc,
...
     rc.bottom = GetMainRect().bottom;
-    rc.left = (rc.left + rc.right) / 2 - width - 1;
+    rc.left = (rc.left + rc.right - width) / 2 - 1;
     rc.right = rc.left + width;
     rc.top = rc.bottom - height;
     break;
 
   case TrafficSettings::GaugeLocation::BOTTOM_RIGHT_AVOID_IB:
     rc.bottom = GetMainRect().bottom;
+	rc.right = GetMainRect().right;
```
Comment (lordfolken) [inline src/MainWindow.cpp:468]: Please fix indent

**Review by lordfolken — CHANGES_REQUESTED** [review]
So just style fixes. 

Please do a git pull --rebase to rebase on current branch so we can avoid the merges.

File: src/MainWindow.cpp (hunk 3 above)
Comment (Joerg-CH) [reply src/MainWindow.cpp:433]: Hope I succeeded now. With "edit in place", every try to indent gave a wrong indentation. With GitHub Desktop, the file edited locally in notepad++ looked fine. Now with I try it with GitHub Dev and the web editor.

File: src/MainWindow.cpp (hunk 2 above)
Comment (Joerg-CH) [reply src/MainWindow.cpp:440]: done.

File: src/MainWindow.cpp (hunk 1 above)
Comment (Joerg-CH) [reply src/MainWindow.cpp:447]: done.

File: src/MainWindow.cpp (hunk 4 above)
Comment (Joerg-CH) [reply src/MainWindow.cpp:468]: done.

**Comment by Joerg-CH** [comment]
Quite difficult to achieve a clean layout, when the editor does not show differences between tab and space. Is there a possibility to show unvisible symbols?

Jörg
schuongf@hotmail.com

**Comment by lordfolken** [comment]
Hmm I tested it, but i don't think it does what i would expect the setting to do:

This is center top (avoid ib)

image

I would expect it to be in the center of the map. 

Testing can be done like this btw:

```
make ./output/UNIX/bin/FeedNMEA
cat test/data/driver/FLARM/pflaf01.nmea | ./output/UNIX/bin/FeedNMEA tcp 4353 
```

**Comment by Joerg-CH** [comment]
You are right, it should be centered in the map window. I did not test unsymmetric IB-layouts, and was happy to find a solution for the (symmetric) landscape layouts. I wonder how to determine the position of the infoboxes, to incorporate this in the calculation of the map window center.

**Comment by Joerg-CH** [comment]
I think I found a solution now. I will update the file MainWindow.cpp tomorrow and commit it to this PR.
image

**Comment by Joerg-CH** [comment]
Latest commit 25e0939 tested with different infobox layouts. Both with "avoid IB" and without. Hope this is OK now.
Test different IB-layouts

**Comment by lordfolken** [comment]
So its good now from a postition point of view. However the size calculation seems to still take the dimentions of GetClientRect() instead of the map, resulting in overlap with the thermal assistant (if configured left/right avoid ib).

**Comment by lordfolken** [comment]
Yes I think thisvis how we should proceed.

**Comment by Joerg-CH** [comment]
I did a mistake: Used the original MainWindow.cpp as basis for the last commit, without the previous changes

**Review by lordfolken — APPROVED** [review]

**Comment by lordfolken** [comment]
image
 Was able to test in in VR. Dark mode doesn't really work in Windows it seems.

**Comment by lordfolken** [comment]
@Joerg-CH i cleaned up the git history, added news and added you to authors. 
Does this look ok for you? Then we could merge it.

**Comment by Joerg-CH** [comment]
OK for me. Thanks for cleaning the history!

**Comment by Joerg-CH** [comment]
@lordfolken 
I am a bit lost, because I do not find anymore the resize of the gauges in the code to avoid overlapping the thermal assistant with the flarm radar. I commited these changes after you answered "Yes I think thisvis how we should proceed."
But strange, I do not find the changes also in my branch 'position-of-flarm-radar'.

**Comment by lordfolken** [comment]
They are all in one commit. I added them as fixup to have one package.

**Comment by lordfolken** [comment]
Wait you may be right. Ill check my got.

**Comment by Joerg-CH** [comment]
I have a local copy of the MainWindow.cpp with ALL my modifications integrated, so it would be not much work to add another commit.

**Comment by lordfolken** [comment]
Please do

**Pull request merged**